tensorflow/java/src/gen/cc/op_gen_main.cc

```diff
@@ -0,0 +1,61 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ==============================================================================*/
+
+#include <string>
+#include <vector>
+
+#include "tensorflow/core/platform/init_main.h"
+#include "tensorflow/core/platform/env.h"
+#include "tensorflow/core/util/command_line_flags.h"
+#include "tensorflow/core/lib/core/status.h"
+#include "tensorflow/core/framework/op.h"
+#include "tensorflow/java/src/gen/cc/op_generator.h"
+
+namespace tensorflow {
+namespace op_gen {
+
+const char kUsageHeader[] =
+    "\n\nGenerator of operation wrappers in Java.\n\n"
+    "This executable generates wrappers for all operations registered in the\n"
+    "ops file it has been linked to (i.e. one of the /core/ops/*.o binaries).\n"
+    "Generated files are output to the path provided as an argument, under\n"
+    "their appropriate package and using a maven-style directory layout.\n\n";
+
+}  // namespace op_gen
+}  // namespace tensorflow
+
+int main(int argc, char* argv[]) {
+  tensorflow::string ops_file;
+  tensorflow::string output_dir;
+  std::vector<tensorflow::Flag> flag_list = {
+    tensorflow::Flag("file", &ops_file,
```

Comment:
The use of the word "file" here seems a bit iffy since the notion of a "file" doesn't quite exist here. It is generating code for all the operations linked in.

Instead of various magic constants embedded in the code, how about we have this tool take in the fully qualified package name as a flag instead? So something like:

- `--output`: Root directory into which output files are generated
- `--package`: Name of the package in which generated ops will be placed

Then all the magic constants can be confined to the genrule implementation, which:
(1) Adds `//tensorflow/core:` prefix to the `op_libs`
(2) Strips `_ops` from `op_libs` and slaps on `org.tensorflow.op` as a prefix for the package name